src/cloudflare/internal/workers.d.ts

```diff
@@ -11,3 +11,10 @@ export class WorkerEntrypoint {
   public ctx: unknown;
   public env: unknown;
 }
+
+export class RpcStub {
+  public constructor(server: object);
+}
+
+export class RpcTarget {
```

Comment:
I thought about it a little bit but it's not clear to me. I think we may want to handle these in different ways. Consider that if you have a DO _stub_ (you're a client of it) and you want to send that stub to someone else over a separate RPC, we probably want to implement that by sending the actor ID, not the live capability. The receiving worker can connect to that actor ID themselves. Similarly if you send a service binding then we can send the worker ID instead of a live capability.

But what happens if you are not a client, but you're doing `return this` inside the object implementation? If we're going to say these things behave like RpcTarget, then `return this` should return a capability to the _specific_ instantiation? But it's weird that this is _different_ from what happens if you send a copy of the binding / DO stub?

This needs more thought and I think we should punt for now.